src/FieldRenderer.js

```diff
@@ -77,7 +77,40 @@ var defaultField;
 var width;
 var counter = 0;
 var Textures;
+var cursor;
 
+var mouseInput = false;
+window.addEventListener("keydown", event => {
+	console.log(event.keyCode);
+	if(event.keyCode == 65){
+		moveViewTo(cursor.getX()-1, cursor.getY());
+		cursor.move(-1,0);
+		mouseInput = false;
+		document.getElementsByTagName("BODY")[0].style.cursor = "none";
+	}else if(event.keyCode == 87){
+		moveViewTo(cursor.getX(), cursor.getY()-1);
+		cursor.move(0,-1);
+		mouseInput = false;
+		document.getElementsByTagName("BODY")[0].style.cursor = "none";
+	}else if(event.keyCode == 83){
+		moveViewTo(cursor.getX(), cursor.getY()+1);
+		cursor.move(0,+1);
+		mouseInput = false;
+		document.getElementsByTagName("BODY")[0].style.cursor = "none";
+	}else if(event.keyCode == 68){
+		moveViewTo(cursor.getX()+1, cursor.getY());
+		cursor.move(1,0);
+		mouseInput = false;
+		document.getElementsByTagName("BODY")[0].style.cursor = "none";
+	}else if(event.keyCode == 32){
+		open();
+		mouseInput = false;
+	}else if(event.keyCode == 16){
+		flag();
+		mouseInput = false;
+	}
```

Comment:
this block needs cleaning. i prefer `switch` statements, the `mouseinput` and `style.cursor` can be moved outside the `if` blocks, and `deltaX` and `deltaY` variables can be used to not repeat the `moveViewTo` and `cursor.move` calls. in the case of `open()` and `flag()`, we can use a `return` to not set the `style.cursor = "none"`